src/pages/login/login.tsx

```diff
@@ -1,16 +1,113 @@
 import React from "react";
 import "./login.css";
 
+interface LoginInput {
+  email: string;
+  password: string;
+}
+
+interface ErrorFormat {
+  error: string;
+  field: string;
+}
+
+const emailRegex = /^\w+([.-]?\w+)*@\w+([.-]?\w+)*(\.\w{3})+$/;
+const passwordRegex = new RegExp(
+  "^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.*[!@#$%^&*])(?=.{7,20})"
+);
+
+function validateFormInputs(input: LoginInput) {
+  if (!input.email) {
+    // return "Email is required.";
+    return { error: "Email is required.", field: "email" };
+  }
+
+  if (!emailRegex.test(input.email)) {
+    // return "Invalid email format.";
+    return { error: "Invalid email format.", field: "email" };
+  }
+
+  if (!input.password) {
+    // return "Password is required.";
+    return { error: "Password is required.", field: "password" };
+  }
+
+  if (!passwordRegex.test(input.password)) {
+    // return "Password has to be at least 7 characters long, one uppercase letter, one lowercase letter, one number and one special character.";
+    return {
+      error:
+        "Password has to be at least 7 characters long, one uppercase letter, one lowercase letter, one number and one special character.",
+      field: "password",
+    };
+  }
+}
+
 export const Login = () => {
+  const [input, setInput] = React.useState<LoginInput>({
+    email: "",
+    password: "",
+  });
+  const [error, setError] = React.useState<ErrorFormat>({
+    error: "",
+    field: "",
+  });
+  const [isSubmitting, setIsSubmitting] = React.useState<boolean>(false);
+
+  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+    const error = validateFormInputs(input);
+
+    if (error) {
+      setError({ error: error.error, field: error.field });
```

Comment:
```suggestion
      setError(error);
```

Vc está copaindo o objeto, então poderia mandá-lo direto, certo? Outro ponto é que vc está chamando o objeto de `error` mas tem um campo `error` dentro dele. Meio estranho, não?

Reply:
Concordo! Corrigido.